Update Readme and Dependencies

adason opened the pull request:
# Peer Review Information

This change includes
- Add tests for python 3.9 
- Update package dependencies 
- Update `README.md` and `CONTRIBUTE.md` files. Remove outdated description.
- Added project status banner 

**The banner looks like:**
<img width="875" alt="Screen Shot 2021-03-30 at 1 44 08 PM" src="https://user-images.githubusercontent.com/786669/113054563-67d68f00-915e-11eb-8a7b-6b57372b6097.png">
 
# Pull Request Check List

<!-- This is just a reminder about the most common mistakes. Please make sure that you tick all *appropriate* boxes. Please read our [contribution guide](https://github.com/Unity-Technologies/dataset-insights/blob/master/CONTRIBUTING.md)
at least once, it will save you unnecessary review cycles! -->

- [ ] Added **tests** for changed code.
- [x] Updated **documentation** for changed code.

Commit:
housekeeping update

Commit:
Add banner

Commit:
fix line sep

Commit:
move status down

Commit:
adjust

Commit:
Add tests for python 3.9 support

Commit:
Merge branch 'cool_banner' into doc_dep_cleanup

Commit:
remove python version mentioned in README

Saurav-D reviewed (dismissed):
Looks good to me. Just that in [Contribute.md](https://github.com/Unity-Technologies/datasetinsights/blob/master/CONTRIBUTING.md#contributing-to-datasetinsights) it says if you want to propose `models/datasets` should we say models here given this repo does not contain any model code now?

Commit:
rewording types of contributions

adason commented:
> Looks good to me. Just that in [Contribute.md](https://github.com/Unity-Technologies/datasetinsights/blob/master/CONTRIBUTING.md#contributing-to-datasetinsights) it says if you want to propose `models/datasets` should we say models here given this repo does not contain any model code now?

Good call. I updated this section and removed model/dataset/evaluation metric completely.

BlairLee commented on .github/workflows/linting-and-unittests.yaml:
```diff
@@ -27,7 +27,7 @@ jobs:
     strategy:
       matrix:
-        python-version: [3.7, 3.8]
+        python-version: [3.7, 3.8, 3.9]
```
Since we support 3.9, we can change it to "We support Python 3 (>=3.7)." in the `Installation` section in [README.md](https://github.com/Unity-Technologies/datasetinsights/blob/master/README.md#installation).

adason replied:
I actually removed it completely. It's obvious from the newly added banner which pulled directly from pypi.

BlairLee commented on pyproject.toml:
```diff
@@ -1,13 +1,13 @@
 [tool.poetry]
 name = "datasetinsights"
-version = "0.2.0"
+version = "1.0.0"
```
Any reason why we want to upgrade the version?

adason replied:
Usually this version does not matter. The pypi package version is based on the release tag which overrides this version flag. I updated it to reduce confusion because we recently released 1.0.
